trimesh/io/misc.py

```diff
@@ -61,129 +62,83 @@ def load_wavefront(file_obj, file_type=None):
+    next_idx = 0
+
+    out_v = []
+    out_vn = []
+    out_vt = []
+    out_f = []
+    g_list = []
+    g_idx = 0
+
+    for line in text.split("\n"):
+        gps = line.strip().split()
+        if len(gps) < 2:
             continue
-
-        # wavefront has 1- indexed faces, as opposed to 0- indexed
-        # additionally, decrement by number of vertices used in prior objects
-        faces = faces.astype(np.int64) - 1 - count_vertices
-
-        # find the data with predictable lengths using numpy
-        data = np.array(text.split())
-        # find the locations of keys, then find the proceeding values
-        # indexes which contain vertex information
-        vid = np.nonzero(data == 'v')[0].reshape((-1, 1)) + np.arange(3) + 1
-        # indexes which contain vertex normal information
-        nid = np.nonzero(data == 'vn')[0].reshape((-1, 1)) + np.arange(3) + 1
-        # some varients of the format have face groups
-        gid = np.nonzero(data == 'g')[0].reshape((-1, 1)) + 1
-
-        # indexes which contain vertex texture information
-        tid = np.nonzero(data == 'vt')[0].reshape((-1, 1)) + np.arange(2) + 1
-
-        loaded = {'vertices': data[vid].astype(float),
-                  'vertex_normals': data[nid].astype(float),
-                  'faces': faces,
-                  'metadata': {}}
-
-        # if face groups have been defined add them to metadata
-        if len(gid) > 0:
-            # indexes which contain face information
-            face_key = np.nonzero(data == 'f')[0]
-            groups = np.zeros(len(faces), dtype=int)
-            for i, g in enumerate(gid):
-                groups[np.nonzero(face_key > g)[0]] = i
-            loaded['metadata']['face_groups'] = groups
-
-        if len(tid) > 0:
-            loaded['metadata']['vertex_texture'] = data[tid].astype(float)
-
-        count_vertices += len(loaded['vertices'])
-
-        loaded_data.append(loaded)
-
-    return list(loaded_data)
+        if gps[0] in attribs: # v, vt, or vn
+            # only parse 3 values-- colors shoved into vertices are ignored
+            attribs[gps[0]].append([float(x) for x in gps[1:4]])
+        elif gps[0] == 'f':
+            ft = gps[1:]
+            if len(ft) == 4:
+                # hasty triangulation of quad
+                ft = [ft[0], ft[1], ft[2], ft[2], ft[3], ft[0]]
+            for f in ft:
+                if not f in remap_table:
+                    remap_table[f] = next_idx
+                    next_idx += 1
+                    gf = f.split('/')
+                    out_v.append(attribs['v'][int(gf[0])-1])
+                    if len(gf) > 1 and gf[1] != '':
+                        out_vt.append(attribs['vt'][int(gf[1])-1])
+                    if len(gf) > 2:
+                        out_vn.append(attribs['vn'][int(gf[2])-1])
+                out_f.append(remap_table[f])
+        elif gps[0] == 'o':
+            _append_mesh(out_v, out_vt, out_vn, out_f, g_list)
+            out_v = []
```

Comment:
After appending a mesh and starting a new one, we're resetting `out_v`, `out_vn`, `out_f`, but not `out_vt`- is that intentional?

Reply:
Oh no, that's a mistake alright.

Reply:
cool fixed in https://github.com/mikedh/trimesh/commit/df89a38d58574a930f7819a486fbe4f133ff1d9e